span-processing-config-service-impl/src/main/java/org/hypertrace/span/processing/config/service/SpanProcessingConfigServiceImpl.java

```diff
@@ -43,21 +52,49 @@
 @Slf4j
 class SpanProcessingConfigServiceImpl
     extends SpanProcessingConfigServiceGrpc.SpanProcessingConfigServiceImplBase {
+
+  private static final String SPAN_PROCESSING_CONFIG_SERVICE = "span.processing.config.service";
+  private static final String SYSTEM_LEVEL_EXCLUDE_SPAN_RULES = "system.level.exclude.span.rules";
+
   private final SpanProcessingConfigRequestValidator validator;
   private final ExcludeSpanRulesConfigStore excludeSpanRulesConfigStore;
   private final TimestampConverter timestampConverter;
   private final ApiNamingRulesManager apiNamingRulesManager;
+  private List<ExcludeSpanRule> systemLevelExcludeSpanRules;
+  private Map<String, ExcludeSpanRuleDetails> systemLevelExcludeSpanRulesToIdMap;
 
   @Inject
   SpanProcessingConfigServiceImpl(
       ExcludeSpanRulesConfigStore excludeSpanRulesConfigStore,
       SpanProcessingConfigRequestValidator requestValidator,
       TimestampConverter timestampConverter,
-      ApiNamingRulesManager apiNamingRulesManager) {
+      ApiNamingRulesManager apiNamingRulesManager,
+      Config config) {
     this.validator = requestValidator;
     this.excludeSpanRulesConfigStore = excludeSpanRulesConfigStore;
     this.timestampConverter = timestampConverter;
     this.apiNamingRulesManager = apiNamingRulesManager;
+
+    buildSystemLevelExcludeSpanRuleConfigs(config);
+  }
+
+  private void buildSystemLevelExcludeSpanRuleConfigs(Config config) {
+    List<? extends com.typesafe.config.ConfigObject> systemLevelExcludeSpanRuleObjectList = null;
+    if (config.hasPath(SPAN_PROCESSING_CONFIG_SERVICE)) {
+      Config spanProcessingConfig = config.getConfig(SPAN_PROCESSING_CONFIG_SERVICE);
+      if (spanProcessingConfig.hasPath(SYSTEM_LEVEL_EXCLUDE_SPAN_RULES)) {
+        systemLevelExcludeSpanRuleObjectList =
+            spanProcessingConfig.getObjectList(SYSTEM_LEVEL_EXCLUDE_SPAN_RULES);
+      }
+    }
```

Comment:
Can we move the configs to a separate config class? Or, may be to a separate method. Would prefer a separate config class, if it isn't much refactor

We can also directly define the default exclusion rules config as `span.processing.config.service.system.exclude.span.rules`, to avoid multiple if block nesting

Reply:
I have added separate methods for reading configs and conversion for now. 
I am thinking of taking up a total refactor(config class, separate manager for exclude rules) separately. What do you suggest?